tests/frontend/upgrade.sh

```diff
@@ -422,3 +423,89 @@ Your packages are up to date.
 	ERROR=""
 	atf_check -o inline:"${OUTPUT}" -e inline:"${ERROR}" -s exit:0 pkg -o REPOS_DIR="$TMPDIR/repoconf" -r ${TMPDIR}/target -o PKG_CACHEDIR="$TMPDIR" -o FORCE_CAN_REMOVE_VITAL=NO upgrade -fy myplop
 }
+
+upgrade_with_dependency_body() {
+	atf_check -s exit:0 sh ${RESOURCEDIR}/test_subr.sh new_pkg "testa" "testa" "1.0"
+	atf_check pkg create -M testa.ucl -o ./repo
+
+	atf_check -s exit:0 sh ${RESOURCEDIR}/test_subr.sh new_pkg "testb" "testb" "1.0"
+	cat << EOF >> testb.ucl
+deps: {
+	testa: {
+		origin: "testa",
+	}
+}
+EOF
+	atf_check pkg create -M testb.ucl -o ./repo
+
+	cat << EOF > pkg.conf
+PKG_DBDIR=${TMPDIR}
+REPOS_DIR=[]
+repositories: {
+	local: { url : file://${TMPDIR}/repo }
+}
+EOF
+	atf_check \
+		-o inline:"Creating repository in ./repo:  done\nPacking files for repository:  done\n" \
+		-e empty \
+		-s exit:0 \
+		pkg -C ./pkg.conf repo ./repo
+
+	atf_check \
+		-o ignore \
+		-s exit:0 \
+		pkg -C ./pkg.conf update -f
+
+	atf_check \
+		pkg -C ./pkg.conf install -qy testb
+
```

Comment:
add rm -r ./repo/* here it will fix your issue, I have created on my side another test which reproduce the issue I will work on fixing it later.